examples/custom-server/src/server.ts

```diff
@@ -0,0 +1,59 @@
+import dotenv from 'dotenv'
+import next from 'next'
+import nextBuild from 'next/dist/build'
+import path from 'path'
+
+dotenv.config({
+  path: path.resolve(__dirname, '../.env'),
+})
+
+import express from 'express'
+import payload from 'payload'
+
+import { seed } from './seed'
+
+const app = express()
+const PORT = process.env.PORT || 3000
+
+const start = async (): Promise<void> => {
+  await payload.init({
+    secret: process.env.PAYLOAD_SECRET || '',
+    mongoURL: process.env.MONGODB_URI || '',
+    express: app,
+    onInit: () => {
+      payload.logger.info(`Payload Admin URL: ${payload.getAdminURL()}`)
+    },
+  })
+
+  if (process.env.PAYLOAD_SEED === 'true') {
+    payload.logger.info('---- SEEDING DATABASE ----')
+    await seed(payload)
+  }
+
+  if (process.env.NEXT_BUILD) {
+    app.listen(PORT, async () => {
+      payload.logger.info(`Next.js is now building...`)
+      // @ts-expect-error
+      await nextBuild(path.join(__dirname, '../'))
+      process.exit()
+    })
+  }
+
+  const nextApp = next({
+    dev: process.env.NODE_ENV !== 'production',
+  })
+
+  const nextHandler = nextApp.getRequestHandler()
```

Comment:
Not sure if typescript 5 related, but I need this line:
```diff
+ process.env.__NEXT_PRIVATE_PREBUNDLED_REACT = 'next'
  const nextHandler = nextApp.getRequestHandler()
```

Reply:
Interesting. I just bumped to TypeScript 5 and not seeing any errors.

Reply:
If I remove this line I get this error: https://github.com/vercel/next.js/issues/48964 (has a comment with this solution)

Reply:
What version of `next` do you have installed?

Reply:
Since this could technically be written as an environment variable, I don't think it should be hard-coded here. This feels like something that Next.js should fix long term. I also wasn't able to recreate. Do I have to use `useRef` somewhere in my code?